examples/model_selection/plot_grid_search_text_feature_extraction.py

```diff
@@ -0,0 +1,235 @@
+print(f"done in {time() - t0:.3f}s")
+
+# %%
+print("Best parameters set:")
+best_parameters = random_search.best_estimator_.get_params()
+for param_name in sorted(parameters.keys()):
+    print(f"{param_name}: {best_parameters[param_name]}")
+
+# %%
+test_accuracy = random_search.score(data_test.data, data_test.target)
+print(
+    "Accuracy of the best parameters using the inner CV of "
+    f"the random search: {random_search.best_score_:.3f}"
+)
+print(f"Accuracy on test set: {test_accuracy:.3f}")
+
+# %%
+# The prefixes `vect` and `clf` are required to avoid possible ambiguities in
+# the pipeline, but are not necessary for visualizing the results. Because of
+# this, we define a function that will rename the tuned hyperparameters and
+# improve the readability.
+
+import pandas as pd
+
+
+def shorten_param(param_name):
+    if "__" in param_name:
+        return param_name.rsplit("__", 1)[1]
+    return param_name
+
+
+cv_results = pd.DataFrame(random_search.cv_results_)
+cv_results = cv_results.rename(shorten_param, axis=1)
+
+# %%
+# We can use a `plotly.express.scatter
+# <https://plotly.com/python-api-reference/generated/plotly.express.scatter.html>`_
+# to visualize the trade-off between scoring time and mean test score. Passing
+# the cursor over a given point displays the corresponding parameters.
+
+import plotly.express as px
+
+param_names = [shorten_param(name) for name in parameters.keys()]
+labels = {"mean_score_time": "score time (s)", "mean_test_score": "CV score"}
+fig = px.scatter(
+    cv_results,
+    x="mean_score_time",
+    y="mean_test_score",
+    error_x="std_score_time",
+    error_y="std_test_score",
+    hover_data=param_names,
+    labels=labels,
+)
+fig
+
+# %%
+# Notice that the cluster of models in the upper-left corner of the plot have
+# the best trade-off between accuracy and scoring time. In this case, using
+# bigrams increases the required scoring time without improving considerably the
+# accuracy of the pipeline. For more information on how to customize an
+# automated tuning to maximize score and minimize scoring time, see the example
+# notebook
+# :ref:`sphx_glr_auto_examples_model_selection_plot_grid_search_digits.py`.
+#
+# We can also use a `plotly.express.parallel_coordinates
+# <https://plotly.com/python-api-reference/generated/plotly.express.parallel_coordinates.html>`_
+# to further visualize the mean test score as a function of the tuned
+# hyperparameters. This helps finding interactions between more than two
+# hyperparameters and provide an intuition on the relevance they have for
+# maximizing the performance of a pipeline.
+
+import math
+
+column_results = param_names + ["mean_test_score", "mean_score_time"]
+
+transform_funcs = dict.fromkeys(column_results, lambda x: x)
+transform_funcs["alpha"] = math.log10
+transform_funcs["norm"] = lambda x: 2 if x == "l2" else 1
+# unigrams are mapped to index 1 and bigrams to index 2
+transform_funcs["ngram_range"] = lambda x: x[1]
+
```

Comment:
We need a bit more inline comment to explain why we need these transform (codewise)